docs/cpp/cpp-refactoring.md

```diff
@@ -0,0 +1,73 @@
+---
+Order: 11
+Area: cpp
+TOCTitle: Refactoring
+ContentId: 5b5da770-b59f-4eca-94a3-78e824d16b52
+PageTitle: Refactoring C++ code
+DateApproved: 1/16/2024
+MetaDescription: How to refactor C++ source files in Visual Studio Code.
+---
+# Refactoring C++ code
+
+The C/C++ Extension in Visual Studio Code has multiple refactoring features to help you improve your code's structure, readability, and maintainability without altering its runtime behavior. These include features such as Extract Method and Create Declaration and Definitions.
+
+## Create declarations or definitions
+
+Simplify the process of creating a function's declaration or definition by letting the C/C++ extension generate these items for you. This functionality applies to member functions, namespaces as classes, and templates.
+
+![Create a definition and a declaration across two files](images/refactoring/create-declaration-and-definition-different-files.gif)
+
+To create a function declaration or definition, either:
+
+* Select your class function definition, then the Code Action (light bulb icon) which opens a dropdown where you can select **Create a Declaration**. For creating a definition, it is the same process, choose the function declaration, then select the Code Action for **Create a Definition**.
+* Right-click a function’s declaration or definition and select the **Create Definition/Declaration** from the context menu. Based on your code, a definition or a declaration will be created.
+* Select the function, then use the **Command Palette** (`kb(workbench.action.showCommands)`) and type in the command **Create Declaration/Definition**. Based on your code, a definition or a declaration will be created.
+
+The location of where the definitions and declarations are created is based on the previous patterns you established in your code. For example, if you previously added definitions and declarations in the same file, the extension will also add any new ones to that same file. Function order is maintained automatically.
+
+![Create a definition and a declaration in the same file](images/refactoring/create-declaration-and-definition-same-file.gif)
+
+If you defined declarations or definitions in a different file from your source file, the extension follows your convention. For example, for a header and a source file with matching names, both are identified, even if the header file has not been included in the source file. In this case, once the declaration or definition is added, your header file is automatically referenced in your source file for you.
+
+If you do not have any precedent of where you create definitions or declarations, we will create a header or source file with a matching name to your current file for you.
```

Comment:
` with a matching name to your current file for you.`->`for you that matches the name of your current file.`